File: Deduction/Proposition/Parsing/RegistryMember.cs
using System;

namespace Deduction.Proposition.Parsing
{
    public class RegistryMember
    {
        // fields
        protected readonly string symbolChar;
        protected readonly Type type;
        protected readonly int precedence;
        protected readonly int parameters;
        protected readonly bool? value;
        protected readonly string[] aliases;
        protected readonly string closesWith;
        protected readonly bool isRightAssociative;

        // constructors
        public RegistryMember(string symbolChar, Type type, int precedence = 10, int parameters = 0, bool? value = null, string[] aliases = null, string closesWith = null, bool isRightAssociative = false)
        {
            this.symbolChar = symbolChar;
            this.type = type;
            this.precedence = precedence;
            this.parameters = parameters;
            this.value = value;
            this.aliases = aliases;
            this.closesWith = closesWith;
            this.isRightAssociative = isRightAssociative;
        }

        // properties
        public string SymbolChar
        {
            get
            {
                return this.symbolChar;
            }
        }

        public Type Type
        {
            get
            {
                return this.type;
            }
        }

        public int Precedence
        {
            get
            {
                return this.precedence;
            }
        }

        public int Parameters
        {
            get
            {
                return this.parameters;
            }
        }

        public bool? Value
        {
            get
            {
                return this.value;
            }
        }

        public string[] Aliases
        {
            get
            {
                return this.aliases;
            }
        }

        public string ClosesWith
        {
            get
            {
                return this.closesWith;
            }
        }

        public bool IsRightAssociative
        {
            get
            {
                return this.isRightAssociative;
            }
        }
    }
}

File: DeductionJava/src/com/eser/ozvataf/deduction/Proposition/Parsing/RegistryMember.java
package com.eser.ozvataf.deduction.Proposition.Parsing;

/**
 * Created with IntelliJ IDEA.
 * User: larukedi
 * Date: 13/01/14
 * Time: 06:18
 * To change this template use File | Settings | File Templates.
 */
public class RegistryMember {
    public String symbolChar;
    public Class<?> type;
    public int precedence;
    public int parameters;
    public Boolean value;
    public String[] aliases;
    public String closesWith;
    public boolean isRightAssociative;

    public RegistryMember() {
    }

    public String getSymbolChar() {
        return this.symbolChar;
    }

    public Class<?> getType() {
        return this.type;
    }

    public int getPrecedence() {
        return this.precedence;
    }

    public int getParameters() {
        return this.parameters;
    }

    public Boolean getValue() {
        return this.value;
    }

    public String[] getAliases() {
        return this.aliases;
    }

    public String getClosesWith() {
        return this.closesWith;
    }

    public boolean isRightAssociative() {
        return this.isRightAssociative;
    }
}
